text/0000-typed-array-conveniences.md

```diff
@@ -0,0 +1,312 @@
+//      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//               ^       ^
+//               |       |
+//              +-------+-------+
+//         arr: |       |       |
+//              +-------+-------+
+//               0       1       2
+let buf = cx.array_buffer(16);
+let arr = JsUint32Array::from_region(&mut cx, buf.region(4, 2))?;
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Convenience types
+
+The `neon::types` module contains the following convenient type aliases:
+
+| Rust Type           | Convenience Type   | JavaScript Type                    |
+| --------------------| ------------------ | ---------------------------------- |
+| `JsTypedArray<u8>`  | `JsUint8Array`     | [`Uint8Array`][Uint8Array]         |
+| `JsTypedArray<i8>`  | `JsInt8Array`      | [`Int8Array`][Int8Array]           |
+| `JsTypedArray<u16>` | `JsUint16Array`    | [`Uint16Array`][Uint16Array]       |
+| `JsTypedArray<i16>` | `JsInt16Array`     | [`Int16Array`][Int16Array]         |
+| `JsTypedArray<u32>` | `JsUint32Array`    | [`Uint32Array`][Uint32Array]       |
+| `JsTypedArray<i32>` | `JsInt32Array`     | [`Int32Array`][Int32Array]         |
+| `JsTypedArray<u64>` | `JsBigUint64Array` | [`BigUint64Array`][BigUint64Array] |
+| `JsTypedArray<i64>` | `JsBigInt64Array`  | [`BigInt64Array`][BigInt64Array]   |
+| `JsTypedArray<f32>` | `JsFloat32Array`   | [`Float32Array`][Float32Array]     |
+| `JsTypedArray<f64>` | `JsFloat64Array`   | [`Float64Array`][Float64Array]     |
+
+These can be convenient to type, and also create a discoverable place for documentation and examples in the API docs.
+
+## Constructing typed arrays
+
+This RFC defines a number of APIs for constructing typed arrays.
+
+### `JsTypedArray`
+
+The `JsTypedArray` type gets three new constructor methods for creating a new typed array.
+
+```rust
+impl<T: Binary> JsTypedArray<T> {
+    // Constructs a new typed array from a length, allocating a new backing buffer.
+    pub fn new<'cx, C>(
+        cx: &mut C,
+        len: usize,
+    ) -> JsResult<'cx, Self>
+    where
+        C: Context<'cx>;
+
+    // Constructs a new typed array from an existing buffer.
+    pub fn from_buffer<'cx, 'b: 'cx, C>(
+        cx: &mut C,
+        buffer: Handle<'b, JsArrayBuffer>,
+    ) -> JsResult<'cx, Self>
+    where
+        C: Context<'cx>;
+
+    // Constructs a new typed array from a region of an existing buffer.
+    pub fn from_region<'c, 'r, C>(
+        cx: &mut C,
+        region: Region<'r, T>,
+    ) -> JsResult<'c, Self>
+    where
+        C: Context<'c>;
+}
+```
+
+### `JsArrayBuffer`
+
+The `JsArrayBuffer` type gets a new method for constructing a `Region` struct, which describes a region of a buffer (see below).
+
+```rust
+impl Handle<'cx, JsArrayBuffer> {
+    pub fn region<T: Binary>(self, byte_offset: usize, len: usize) -> Region<'cx, T>;
+}
+```
+
+### Helper types
+
+#### `neon::types::binary::Binary`
```

Comment:
I think it would be better to put `Binary` and `Region` in the existing `neon::types::buffer` module to keep the tree slightly smaller and to co-locate array buffer related traits/types.

https://docs.rs/neon/latest/neon/types/buffer/index.html

Reply:
Whoops, that was just a brain-o when I typed up the RFC (the implementation PR already agrees with your suggestion). I'll fix the doc.